crates/bevy_color/src/hsla.rs

```diff
@@ -127,91 +127,85 @@ impl Luminance for Hsla {
     }
 }
 
-impl From<Srgba> for Hsla {
+impl From<Hsla> for Hsva {
     fn from(
-        Srgba {
-            red,
-            green,
-            blue,
+        Hsla {
+            hue,
+            saturation,
+            lightness,
             alpha,
-        }: Srgba,
+        }: Hsla,
     ) -> Self {
-        // https://en.wikipedia.org/wiki/HSL_and_HSV#From_RGB
-        let x_max = red.max(green.max(blue));
-        let x_min = red.min(green.min(blue));
-        let chroma = x_max - x_min;
-        let lightness = (x_max + x_min) / 2.0;
-        let hue = if chroma == 0.0 {
-            0.0
-        } else if red == x_max {
-            60.0 * (green - blue) / chroma
-        } else if green == x_max {
-            60.0 * (2.0 + (blue - red) / chroma)
+        let value = lightness + saturation * lightness.min(1. - lightness);
+        let saturation = if value == 0. {
+            0.
         } else {
-            60.0 * (4.0 + (red - green) / chroma)
-        };
-        let hue = if hue < 0.0 { 360.0 + hue } else { hue };
-        let saturation = if lightness <= 0.0 || lightness >= 1.0 {
-            0.0
-        } else {
-            (x_max - lightness) / lightness.min(1.0 - lightness)
+            2. * (1. - (lightness / value))
         };
 
-        Self::new(hue, saturation, lightness, alpha)
+        Hsva::new(hue, saturation, value, alpha)
     }
 }
 
-impl From<Hsla> for Srgba {
+impl From<Hsva> for Hsla {
     fn from(
-        Hsla {
+        Hsva {
             hue,
             saturation,
-            lightness,
+            value,
             alpha,
-        }: Hsla,
+        }: Hsva,
     ) -> Self {
-        // https://en.wikipedia.org/wiki/HSL_and_HSV#HSL_to_RGB
```

Comment:
We should avoid losing this link (or replace it with another derivation if needed).

Reply:
Good point! I'll update all 6 core `From` implementations across HSL/HSV/HWB to explicitly include links to where these formulae come from.

Reply:
You're the best :)